### PR TITLE
Mobile view

### PR DESCRIPTION
**Pull request by Melisa-Zorraindo**
I've adapted the layout to smaller screens. The breakpoint is set at 935px and works well until ~320px.
I've also modified the original layout to feature narrower navigation bars, as they were taking up too much screen.
I've added scrollbar styles that were previously merged into the main branch.

### Summary by OpenAI

Release Notes:

- New Feature: Added new functions `getCurrent`, `getEntityMeta`, `getEntitySlug`, `getEntityChildren`, `getEntityFrontmatter`, and modified `parseMarkdown` to return an array of objects with arrays of children instead of a nested object.
- Style: Adapted the layout to smaller screens, modified the original layout to feature narrower navigation bars, and added scrollbar styles. It also handles window resizing and toggles for recursive and course navigation bars. Additionally, `institutes` data structure has been changed from an object to an array in `ProgrammeNav.svelte`.
- Documentation: Added new content for the website "Hackademics" that offers online courses in coding and web development.

> A new layout for small screens,
> With narrower navs and scrollbars clean,
> Functions added, markdown parsed anew,
> Hackademics content for learners true.

### CODE REVIEW

File: packages/lms/src/lib/components/content/Main.svelte [hunk 1]
```diff
@@ -12,5 +12,32 @@
...
+		background-color: var(--primary);
+		border-radius: var(--border-radius-xs);
+	}
+
+	@media screen and (max-width: 935px) {
+		main {
+			overflow: inherit;
+			grid-area: 3 / 1 / 4 / 5;
+			padding-bottom: 3rem;
+		}
```
Comment (pretzL) [inline packages/lms/src/lib/components/content/Main.svelte:41]: I remember seeing something in the docs/project tasks about preferring container queries over media queries. Did we step away from this?

File: packages/lms/src/lib/components/footer/Footer.svelte [hunk 2]
```diff
@@ -11,4 +11,14 @@
...
+
+	@media screen and (max-width: 935px) {
+		footer {
+			grid-area: 4 / 1 / 5 / 5;
+			position: fixed;
+			bottom: 0;
+			width: 100%;
+			height: 3rem;
+		}
+	}
```
Comment (pretzL) [inline packages/lms/src/lib/components/footer/Footer.svelte:23]: mediaaaa

File: packages/lms/src/lib/components/header/Logo.svelte [hunk 3]
```diff
@@ -12,4 +12,10 @@
 	h1 {
 		margin: auto;
 	}
+
+	@media screen and (max-width: 935px) {
+		a {
+			grid-area: 1 / 1 / 2 / 2;
+		}
+	}
```
Comment (pretzL) [inline packages/lms/src/lib/components/header/Logo.svelte:20]: video killed the media star?

File: packages/lms/src/lib/components/header/Settings.svelte [hunk 4]
```diff
@@ -47,7 +47,7 @@
 
 <style>
 	.wrapper {
-		grid-area: settings;
+		/* grid-area: settings; not doing anything*/
```
Comment (pretzL) [inline packages/lms/src/lib/components/header/Settings.svelte:50]: If something doesn't do anything, it's probably best to just remove it?

File: packages/lms/src/lib/components/navigation/CourseNav.svelte [hunk 5]
```diff
@@ -31,22 +36,27 @@
 		font-weight: 600rem;
```
Comment (pretzL) [inline packages/lms/src/lib/components/navigation/CourseNav.svelte:36]: Didn't this get fixed somewhere or am I drunk?

File: packages/lms/src/lib/components/navigation/IconNav.svelte [hunk 6]
```diff
@@ -32,4 +32,10 @@
 	i {
 		height: min-content;
 	}
+
+	@media screen and (max-width: 935px) {
+		.wrapper {
+			grid-area: 2 / 1 / 3 / 2;
+		}
+	}
```
Comment (pretzL) [inline packages/lms/src/lib/components/navigation/IconNav.svelte:40]: Another one bites the dust

File: packages/lms/src/lib/components/navigation/CourseNav.svelte (hunk 5 above)
Comment (Anclagen) [reply packages/lms/src/lib/components/navigation/CourseNav.svelte:36]: Yeah got fixed on content-dependencies, this probably just need the master merge into it to update

File: packages/lms/src/lib/components/content/Main.svelte (hunk 1 above)
Comment (Fermain) [reply packages/lms/src/lib/components/content/Main.svelte:41]: Nope, I would like to enforce @container queries throughout unless @media is utterly unavoidable

**Comment by Melisa-Zorraindo** [comment]
I have replaced the MQ at component level by CQ.
I have also changed the break point to `870px`.

File: packages/lms/src/lib/components/header/Settings.svelte (hunk 4 above)
Comment (Melisa-Zorraindo) [reply packages/lms/src/lib/components/header/Settings.svelte]: That was a comment for myself that I forgot to remove. Done it now.

File: packages/lms/src/lib/components/content/Main.svelte (hunk 1 above)
Comment (Jenfer93) [reply packages/lms/src/lib/components/content/Main.svelte]: This works fine for me, but think the main content should have some margin left and right as it's kind of hugging the edges as the screen gets smaller

Comment (Melisa-Zorraindo) [reply packages/lms/src/lib/components/content/Main.svelte]: I've added some padding now

**Review by Fermain — APPROVED** [review]
As a prototype this is good to go for me. Next sprint we will be implementing Skeleton and redoing much of this but it serves as a good basis for that.

Things to think about next sprint:
- Do we like this layout?
- Should the reader stay in the footer?
- What should we do with the icon nav?

**Pull request merged**